### ISSUE TITLE
support transformation functions

### ISSUE THREAD

**Issue by dmitryikh** [dmitryikh/leaves#23]
Currently _leaves_ outputs prediction as a raw score. Client code should transform it to probabilities (logistic), lambdarank and so.

Let's introduce this ability to _leaves_.

Short list for XGBoost:
  * "binary:logistic"
  * "binary:logitraw"
  * "multi:softmax"
  * "multi:softprob"
  * "reg:linear"

Short list for LightGBM:
todo

**Comment by dmitryikh** [comment]
partially done in #54 

but not all transformation functions were implemented. Currently here are logistic & softmax transformations for Lightgbm and XGBoost.

Close for now
